### ISSUE TITLE
deserializeStream crashes!!!!!! (the `corrupt bson message' error throw is wrong)

### ISSUE THREAD

**Issue by JoelParke** [mongodb/js-bson#194]
I can't really believe it, but it seems that a check in deserializer.js causes 'corrupt bson message' to be thrown.   I can't quite believe it, since this has been around for quite a while. So I added a couple of console.log() messages:_

```
// Ensure buffer is valid size
  if(size < 5 || buffer.length < size || (size + index) < buffer.length) {
                console.log(size, buffer.length, size+index);
                console.log(size < 5, buffer.length < size, (size+index) < buffer.length);
		throw new Error("corrupt bson message");
	}
```
I see: `37 36000 37` and `false false true`.  VERIFYING that the last check is BACKWARDS.  So I changed the code to:

```
	// Ensure buffer is valid size
  if(size < 5 || buffer.length < size || (size + index) > buffer.length) {
		throw new Error("corrupt bson message");
	}
```
and all works correctly.  I checked this back through 0.5.0 and it is broken, unless I totally am confused.  Thanks!

**Comment by JoelParke** [comment]
Add a PR #195  to fix this.

**Comment by christkv** [comment]
merged and released as 0.5.6
